Source/bmalloc/libpas/src/libpas/pas_report_crash.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PAS_REPORT_CRASH_H
+#define PAS_REPORT_CRASH_H
+
+#include "pas_utils.h"
+
+#include "pas_ptr_hash_set.h"
+#include "pas_ptr_hash_map.h"
+#include "pas_root.h"
+
+#if PAS_OS(DARWIN)
+#include <mach/mach_types.h>
+#include <mach/vm_types.h>
+#endif
+
+PAS_BEGIN_EXTERN_C;
+
+#if PAS_OS(DARWIN)
+// Read memory from crashed process.
+typedef void *(*crash_reporter_memory_reader_t)(task_t task, vm_address_t address, size_t size);
+
+// Report sent back to the ReportCrash process.
+typedef struct {
+    const char *error_type;
+    const char *confidence;
+    vm_address_t fault_address;
+    size_t allocation_size;
+} pas_report_crash_pgm_report;
```

Comment:
Since this struct is being exposed as part of the part of the JSC SPI, I would have preferred that it be define in the JSC SPI header file.  However, that would introduce a layer violation.  I suppose we can live with this.

More importantly, we should define a version field at the top of the struct.  `pas_report_crash_extract_pgm_failure` should fill in this `version` so that the client can check if it knows how to parse the contents of `pas_report_crash_pgm_report`.

Reply:
Added version as part of the function